Download.Mvc/Client/components/FilePreview.tsx

```diff
@@ -0,0 +1,35 @@
+﻿import * as React from 'react';
+import 'isomorphic-fetch';
+
+export class FilePreview extends React.Component<any, any>{
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            link: '',
+        };
+    }
+    componentDidMount = () => {
+        // grab link to download
+        if (this.props.id) {
+            fetch('/api/previewFile/' + this.props.id, { credentials: 'same-origin' })
+                .then(response => response.json())
+                .then(link => this.setState({ link }));
```

Comment:
@jpknoll is there a reason you recommend this for Preview but not any other time I'm using fetch? 
also, catching looks like: 
```
      .catch((error) => {
        console.error(error);
      });
```
right?